test/test_dtype_alu.py

```diff
@@ -93,6 +93,8 @@ def universal_test_midcast(a, b, c, op1, op2, d1:DType, d2:DType):
   np.testing.assert_allclose(tensor_value, numpy_value, rtol=1e-6 if getenv("PTX") else 1e-7)
 
 class TestDTypeALU(unittest.TestCase):
+  def setUp(self): np.errstate(all='ignore').__enter__() # context manager for floating-point error handling
```

Comment:
Are you sure this is cleaned up? Can you just put the decorator on the class?

Reply:
Wow, I didn't know classes could be wrapped! I was so focused on wrapping functions that I didn't consider that possibility. I will do better.